src/Symfony/Component/Messenger/Transport/Serialization/Serializer.php

```diff
@@ -67,14 +69,49 @@ public function encode(Envelope $envelope): array
             $context = $serializerConfig->getContext() + $context;
         }
 
-        $headers = array('type' => \get_class($envelope->getMessage()));
-        if ($configurations = $envelope->all()) {
-            $headers['X-Message-Envelope-Items'] = serialize($configurations);
-        }
+        $headers = array('type' => \get_class($envelope->getMessage())) + $this->encodeEnvelopeItems($envelope);
 
         return array(
             'body' => $this->serializer->serialize($envelope->getMessage(), $this->format, $context),
             'headers' => $headers,
         );
     }
+
+    private function decodeEnvelopeItems($encodedEnvelope)
+    {
+        if ($this->envelopeWithSerialize) {
```

Comment:
should this flag be set? isn't `isset($encodedEnvelope['headers']['X-Message-Envelope-Items'])` enough?

Reply:
You are right, it would be better 👌

Reply:
Updated 👍